### ISSUE TITLE
I need help

### ISSUE THREAD

**Issue by ikke68** [davolesh/mqtt_pushbutton#1]
Hi Dave,

I saw your video on youtube, and i flashed a NodeMCU with your software using arduino. That all worked out because my device is connected to my wifi, but how do i set up my device in Home Assistant? because for other projects i use tasmota, and tasmota got a auto discovery function..
I'm looking forward to your answer!!
Greatings, Rens

**Comment by OleshMC** [comment]
Hi and thanks for getting in touch.

To get this to work with home assistant, you need to set up two different automations. 1 automation for the quick press, in one for the long press.

Here is an example of one of the automations. The automation wait for the word "toggle" then flips the switch of your choice.

![Screenshot_20200225-071807_Home Assistant~01](https://user-images.githubusercontent.com/61282966/75247092-29bead00-579f-11ea-82d2-1096de6c21bc.jpg)
![Screenshot_20200225-071801_Home Assistant~01](https://user-images.githubusercontent.com/61282966/75247109-304d2480-579f-11ea-8dcf-8675e1b2543d.jpg)

I actually have this set up with tasmota in my house. Both click and hold work for two different lights. Auto discovery also works. There will be a video on this soon.

-David

**Comment by davolesh** [comment]
Hi and thanks for getting in touch.

To get this to work with home assistant, you need to set up two different automations. 1 automation for the quick press, in one for the long press.

Here is an example of one of the automations. The automation wait for the word "toggle" then flips the switch of your choice.

![Screenshot_20200225-071801_Home Assistant~01](https://user-images.githubusercontent.com/10506816/75247297-99cd3300-579f-11ea-88c9-0268a1fb62e7.jpg)
![Screenshot_20200225-071807_Home Assistant~01](https://user-images.githubusercontent.com/10506816/75247312-a05baa80-579f-11ea-8e45-626f0c6270a5.jpg)

I actually have this set up with tasmota in my house now. Both click and hold work for two different lights. Auto discovery also works. There will be a video on this soon.

-David

**Comment by ikke68** [comment]
Hi David,

Thanks for the fast reply! i'll try it with your pictures.
Is it possible you can show me now how you set it up in tasmota? Because that easier for me, i have a few tasmotized device's and that way i keep it al the same :)
Thanks Rens,

**Comment by davolesh** [comment]
Absolutely, I'll give you the commands needed tonight, (they are at home, I'm at work.)

**Comment by ikke68** [comment]
Okay thanks allot!

**Comment by ikke68** [comment]
for the mqtt server part do i need to put :8123 behind it?

**Comment by davolesh** [comment]
8123 is the home assistant port. The MQTT port is 1883.
